.changeset/good-grapes-double.md

```diff
@@ -0,0 +1,5 @@
+---
+'lit-html': patch
+---
+
+Improved the type inferece of the `choose()` directive to properly restrict the case type inferred from provided value.
```

Comment:
Adding some additional info for potential user breakage as suggested by @AndrewJakubowicz 
```suggestion
Improved the type inferece of the `choose()` directive to properly restrict the case type inferred from provided value. **Note**: If this change creates a type error in your code, there must have been an unreachable case that can be removed, or the type of your `value` might be missing a valid case in the union.
```

Reply:
Beautiful!